bcipy/tasks/rsvp/copy_phrase.py

```diff
@@ -44,368 +71,610 @@ class RSVPCopyPhraseTask(Task):
+                self.update_session_data(data,
+                                         save=True,
+                                         decision_made=decision.decision_made)
+
+                if decision.decision_made:
+                    self.show_feedback(decision.selection,
+                                       (decision.selection == target_letter))
+                    self.spelled_text = decision.spelled_text
+                    self.current_inquiry = self.next_inquiry()
+
+                else:
+                    self.current_inquiry = decision.new_inq_schedule
+
+                run = self.check_stop_criteria()
+                self.inq_counter += 1
+
+            self.exit_display()
+            self.write_offset_trigger(trigger_file)
 
         # Wait some time before exiting so there is trailing eeg data saved
-        core.wait(self.eeg_buffer)
+        self.wait(seconds=self.parameters['eeg_buffer_len'])
 
         return self.file_save
 
-    def name(self):
+    def evaluate_evidence(self) -> Decision:
+        """Uses the `copy_phrase_task` parameter to evaluate the provided
+        evidence and attempt a decision. 
+        
+        Modifies
+        --------
+        - self.copy_phrase_task
+        """
+        if self.fake:
+            _, spelled, _ = fake_copy_phrase_decision(self.copy_phrase,
+                                                      self.next_target(),
+                                                      self.spelled_text)
+            # Reset the stoppage criteria by forcing the commit to a decision.
+            self.copy_phrase_task.decision_maker.do_series()
+
+            # In fake mode, all inquiries result in a selection.
+            return Decision(decision_made=True,
+                            selection=spelled[-1],
+                            spelled_text=spelled,
+                            new_inq_schedule=None)
+
+        decision_made, new_sti = self.copy_phrase_task.decide()
+        spelled_text = self.copy_phrase_task.decision_maker.displayed_state
+        selection = self.copy_phrase_task.decision_maker.last_selection
+
+        return Decision(decision_made, selection, spelled_text, new_sti)
+
+    def add_evidence(self, stim_times: List[List],
+                     proceed: bool = True) -> List[str]:
+        """Add all evidence used to make a decision.
+
+        Parameters
+        ----------
+        - stim_times : list of stimuli returned from the display
+        - proceed : whether or not to proceed with the inquiry
+        
+        Returns
+        -------
+        list of evidence types added
+
+        Modifies
+        --------
+        - self.copy_phrase_task
+        """
+        evidences = [
+            self.compute_button_press_evidence(proceed),
+            self.compute_eeg_evidence(stim_times, proceed)
+        ]
+        evidence_types = []
+        for evidence in evidences:
+            if evidence:
```

Comment:
I like this approach - all the evidences can be included and simply set to None if not desired. 
It would be good to have tests that check the alphabet distribution in situations like (have inquiry preview, don't have EEG), (no preview, have EEG), (have preview, have EEG) - and see that the changes to letter probabilities are as expected.
This might already be covered by some simple integration tests of copy_phrase_wrapper or of this task - but just wanted to mention it here while I think of it.

Reply:
In particular, I mention it because the test might have to do some "under-the-hood" inspection (like checking `task.copy_phrase_task.symbol_probabilities` (or whatever the correct attribute is, rather than `symbol_probabilities`).

On the other hand, this would feel a bit strange (tests of `copy_phrase` probably should not know/care about its attributes like `self.copy_phrase_task` too much).

Reply:
Yeah, I think the tests for the Copy Phrase task should just confirm that the add_evidence method was called with the expected types, and the CopyPhraseWrapper should assert that the shape of the provided evidence is correct. Whenever possible I tried to limit the level of knowledge that the Copy Phrase task has about the implementation details of the wrapper.

Reply:
Ok - sounds like a reasonable way to divide the unit tests